repo_resource/common.py

```diff
@@ -379,6 +380,20 @@ def update_manifest(self, jobs):
                     defaultBranch = defaults.get('revision') \
                         or self.__get_remote_revision(defaultRemote)
 
+                # Handle <remove-project /> element:
+                # - Should not be present in final XML (for version)
+                # - Should remove first matching project by name
+                for p in manifest.findall('remove-project'):
+                    project = p.get('name')
+                    removed_projects.append(project)
+                    manifest.remove(p)
+
+                for p in manifest.findall('project'):
```

Comment:
This could go in the next block, couldn't it? I guess that's for code clarity?
I know it's not not very time consuming, but I think it'd be worth checking remove_projects isn't empty before the block, and after each iteration, since there are usually many more projects items than projects removed.

Reply:
I'm not sure it can go in the next code block.

This loop actually modifies the `manifest` Tree and I don't know how `findall()` behaves:
- If it takes a snapshot at the beginning of the loop, then we will end up parsing an unwanted project anyways.

I also prefer to keep it separate for clarity.
If you prefer, I can move both paragraph together (without a blank line between both).

Reply:
For v2, I will add the following check:

```python
                for p in manifest.findall('project'):
                    project = p.get('name')
                    if project in removed_projects:
                        manifest.remove(p)
                        removed_projects.remove(project)

                     # If there are no more projects to remove, we can 
                     # skip parsing the rest of the projects
                     if removed_projects == []:
                          break  
```

I honestly don't think that it matters much, performance wise (because I have not measured it) but it's simple enough to add